.github/workflows/main.yml

```diff
@@ -0,0 +1,35 @@
+name: CI
+
+on:
+  push:
+    branches: ["master"]
+  pull_request:
+    branches: ["master"]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12", "3.13"]
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run tests
+        run: uv run --with python-dateutil --with six test.py
```

Comment:
Something is odd -- tests are failing, but appear to pass ....

And can we use pytest ? -- only 'cause I like it -- it give far more useful info on failure. I think that would take somethign like:

`run: uv run --with python-dateutil --with six --with pytest pytest --pyargs bson`

though I don't really know uv ...

Reply:
Where's it failing?

Reply:
I looked at the remove py2 PR, and is showed as passing, but it was failing, at least on py3.9.

https://github.com/py-bson/bson/actions/runs/12401626827/job/34621412735

But I've fixed that now -- so it all passes now. 

(I also changed the test run to pytest in the workflow file in that PR -- maybe shouldn't have mixed and matched , but ...)